### PR TITLE
Fixed drone dispenser material exploit

### PR DESCRIPTION
**Pull request by xxalpha**
Fixes https://github.com/tgstation/-tg-station/issues/17442

:cl:
bugfix: Fixed drone dispenser multiplying glass or metal.
/:cl:

### CODE REVIEW

File: code/game/machinery/droneDispenser.dm [hunk 1]
```diff
@@ -39,25 +39,29 @@
...
 /obj/machinery/droneDispenser/Destroy()
 	SSmachine.processing -= src
 	return ..()
 
-/obj/machinery/droneDispenser/preloaded
-	metal = 5000
-	glass = 5000
+/obj/machinery/droneDispenser/preloaded/New()
+	..()
+	materials.insert_amount(5000)
```
Comment (coiax) [inline code/game/machinery/droneDispenser.dm]: So this'll put in 5000 of metal *and* glass?

Comment (xxalpha) [inline code/game/machinery/droneDispenser.dm]: That's how it works.

File: code/game/machinery/droneDispenser.dm [hunk 2]
```diff
@@ -39,25 +39,29 @@
 /obj/machinery/droneDispenser/New()
 	..()
 	health = max_health
+	materials = new(src, list(MAT_METAL=1, MAT_GLASS=1), MINERAL_MATERIAL_AMOUNT*MAX_STACK_SIZE*2)
+	using_materials = list(MAT_METAL=metal_cost, MAT_GLASS=glass_cost)
 	SSmachine.processing |= src
 
 /obj/machinery/droneDispenser/Destroy()
 	SSmachine.processing -= src
```
Comment (phil235) [inline code/game/machinery/droneDispenser.dm]: I'm not sure but don't we need to qdel datum references? like we do qdel(reagents) for reagent containers.

Comment (coiax) [inline code/game/machinery/droneDispenser.dm]: I think the garbage collector picks them up eventually, but it's better if we remove them when we get Destroyed

Comment (xxalpha) [inline code/game/machinery/droneDispenser.dm]: yea

**Pull request merged**